### PR TITLE
PORTAL-1538: add new governmentInfo fields used for vn

### PR DESCRIPTION
**Pull request by daintablian**

### CODE REVIEW

File: src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json [hunk 1]
```diff
@@ -580,6 +580,49 @@
...
+					"type": "string",
+					"pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$",
+					"description": "Date when the customer received training (YYYY-MM-DD). Used for both customer and coapplicant. VN only."
+				},
+				"cardDate": {
+					"type": "string",
+					"pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$",
+					"description": "Date associated with the customer's card (YYYY-MM-DD). Used for both customer and coapplicant. VN only."
+				},
+				"trainerName": {
```
Comment (bluetechy) [inline src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json:611]: @daintablian The API has consistently followed this object model for names: https://github.com/Unicity/PublicModels.json/blob/master/src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Order.json#L146-L149

File: src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json [hunk 2]
```diff
@@ -580,6 +580,49 @@
...
+		},
+		"governmentInfo": {
+			"type": "object",
+			"description": "Government-related information for the customer",
+			"properties": {
+				"governmentIdPlace": {
+					"type": "string",
+					"description": "Place where the government ID was issued. Used for both customer and coapplicant. VN only."
+				},
+				"idIssuedDate": {
```
Comment (bluetechy) [inline src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json:592]: @daintablian The API has consistently followed the naming convention date*******. For example: https://github.com/Unicity/PublicModels.json/blob/master/src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Order.json#L39-L62

**Review by bluetechy — CHANGES_REQUESTED** [review]

File: src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json (hunk 1 above)
Comment (daintablian) [reply src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json:616]: Am I know following what here is not the object model?

File: src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json [hunk 3]
```diff
@@ -580,6 +580,49 @@
...
+					"type": "string",
+					"pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$",
+					"description": "Date when the government ID was issued (YYYY-MM-DD). Used for both customer and coapplicant. VN only."
+				},
+				"contractNumber": {
+					"type": "string",
+					"description": "Contract number associated with the customer. Used for both customer and coapplicant. VN only."
+				},
+				"trainDate": {
```
[the remark this answers is not in the document]
Comment (daintablian) [reply src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json]: done

File: src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json [hunk 4]
```diff
@@ -580,6 +580,49 @@
...
+					"type": "string",
+					"pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$",
+					"description": "Date when the customer received training (YYYY-MM-DD). Used for both customer and coapplicant. VN only."
+				},
+				"cardDate": {
```
[the remark this answers is not in the document]
Comment (daintablian) [reply src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json]: done

File: src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json [hunk 5]
```diff
@@ -580,6 +580,49 @@
...
+				"trainerName": {
+					"type": "string",
+					"description": "Name of the trainer who provided training to the customer. Used for both customer and coapplicant. VN only."
+				},
+				"memberCard": {
+					"type": "string",
+					"pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$",
+					"description": "Date Member Card was given to the coapplicant. Used ONLY for coapplicant. VN only."
+				},
+				"trainingCardDate": {
```
[the remark this answers is not in the document]
Comment (daintablian) [reply src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json]: done

File: src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json (hunk 2 above)
Comment (daintablian) [reply src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json]: done

File: src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json (hunk 1 above)
Comment (bluetechy) [reply src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json:616]: @daintablian Originally, we had this object model called `identification`, but that was used in markets where they had more than one government id.  This looks strangely like that object: https://github.com/Unicity/PublicModels.json/blob/master/src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json#L270-L302

What is your use case?

Comment (daintablian) [reply src/classes/Unicity/MappingService/Impl/Hydra/API/Master/Model/Customer.json:616]: this is for the vietnam specific fields. you can see the hydra key, unity key, and xdbc key [here](https://github.com/Unicity/Octopus/blob/release/components/services/hydra-app/src/classes/v5/Helper/VNGovernmentInfoHelper.php#L58) , if that gives any context.

**Pull request merged**